usaspending_api/references/v2/views/cfda.py

```diff
@@ -26,15 +29,19 @@ def get(self, request, cfda=None):
             result = CFDA_DICTIONARY.get(cfda)
 
             if not result:
-                return Response(status=404)
-
-            response = {
-                "cfda": result["cfda"],
-                "posted": result["posted"],
-                "closed": result["closed"],
-                "archived": result["archived"],
-                "forecasted": result["forecasted"],
-            }
+                raise NoDataFoundException(f"No grant records found for '{cfda}'.")
+
+            try:
+                response = {
+                    "cfda": result["cfda"],
+                    "posted": result["posted"],
+                    "closed": result["closed"],
+                    "archived": result["archived"],
+                    "forecasted": result["forecasted"],
+                }
+            except KeyError:
+                raise InternalServerError(f"Data from {CFDA_URL} not in expected format: {result}")
```

Comment:
I discourage returning a HTTP 500 error. I definitely would not publicly return the CFDA URL and response in our endpoint response. The main purpose of this ticket was A) better logging and B) robust client code to handle potential problems accessing the external service